go/libraries/doltcore/merge/schema_merge_test.go

```diff
@@ -1083,6 +1083,20 @@ var secondaryIndexTests = []schemaMergeTest{
 		skipOldFmt:          true,
 		skipFlipOnOldFormat: true,
 	},
+	{
+		name:     "dropping column on right shifts column index between compatible types (see pr/8154)",
```

Comment:
It looks like the `incompatible types` test case below errors out with a panic without the fix for this issue, but the `compatible types` test case seems to still pass when I ran it against main. 

Got any ideas on a better way to test the correctness of the updates to the secondary index that would catch the data issue even though this one was able to merge correctly? I had thought the code in `ValidateDatabase()` / `validateSecondaryIndexes` would catch it, but it doesn't seem to be running for these tests. Turning that validation on for these tests could be one easy way to catch that. 

Might also be nice to cover insert/update/delete cases with the secondary index update if we can, too.

Reply:
Fixed with a new test `dolt_queries_schema_merge`